python/pfs/drp/stella/fitBroadbandSED.py

```diff
@@ -0,0 +1,135 @@
+from lsst.pex.config import Config, Field, DictField
+from lsst.pipe.base import Task
+from lsst.utils import getPackageDir
+
+import numpy
+import numpy.lib.recfunctions
+import astropy.io.fits
+
+import os
+
+
+class FitBroadbandSEDConfig(Config):
+    """Configuration for FitBroadbandSEDTask
+    """
+
+    fluxLibraryPath = Field(
+        dtype=str,
+        default=os.path.join(getPackageDir("drp_pfs_data"), "fluxCalibration",
+                             "syntheticFlux_withInterp_ps_HSCPS1GaiaSDSS.fits"),
+        doc="Synthetic photometry table"
+    )
+
+    filterAliases = DictField(
+        keytype=str, itemtype=str,
+        default={
+            "g": "HSCg", "r": "HSCr", "r2": "HSCr2", "i": "HSCi", "i2": "HSCi2",
+            "z": "HSCz", "y": "HSCy"
+        },
+        doc="Conversion table from AFW's filter names to those used by `fluxLibrary`"
+    )
+
+
+class FitBroadbandSEDTask(Task):
+    """Fit an observed SED with synthetic model SEDs
+    """
+
+    ConfigClass = FitBroadbandSEDConfig
+    _DefaultName = "fitBroadbandSED"
+
+    def runDataRef(self, dataRef):
+        """For each spectrum,
+        calculate probabilities of model SEDs matching the spectrum.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.ButlerDataRef`
+            Data reference for merged spectrum.
+
+        Returns
+        -------
+        broadbandPDF : `list` of `numpy.array` of `float`
+            ``broadbandPDF[iSpectrum][iSED]`` is the normalized probability
+            of the SED ``iSED`` matching the spectrum ``iSpectrum``.
+        """
+        pfsConfig = dataRef.get("pfsConfig")
+
+        pdfs = []
+        for filterNames, fiberFlux, fiberFluxErr \
+                in zip(pfsConfig.filterNames, pfsConfig.fiberFlux, pfsConfig.fiberFluxErr):
+            pdfs.append(self.run(filterNames, fiberFlux, fiberFluxErr))
+
+        return pdfs
+
+    def run(self, filterNames, fiberFlux, fiberFluxErr):
```

Comment:
Can we have the `run` method simply take a `pfsConfig`? That makes it easier for users to call directly.

Reply:
`pfsConfig` contains multiple spectra, but `run()` here deals with one spectra per call. Should I change `run()` so it will take `pfsConfig`, and rename the old `run()` to `getProbabilities()`?

Reply:
Yeah, that sounds good!